### PR TITLE
outcome_transform ignored for ModelListGP.fantasize

### PR DESCRIPTION
**Pull request by esantorella**
This constitutes a very minimal solution to #1333 that I mean to improve. It could be tested more thoroughly and efficiently, and I (or any helpful reviewers :) ) should spend a little more time thinking about the logic.

## Motivation

#1333 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

[x] Test that with a large number of samples, the posterior mean of the original models is close to the average posterior mean of fantasy models.
[ ] Make this test more comprehensive -- try different dtypes and different types of `ModelListGP` component models
[x] Make this test more efficient -- ideally it shouldn't need such a large number of samples

### CODE REVIEW

File: test/models/test_model_list_gp_regression.py [hunk 1]
```diff
@@ -312,3 +314,31 @@ def test_fantasize(self):
...
+                    for i in range(Y.shape[-1])
+                ]
+                model = ModelListGP(*models_1d)
+                model.eval()
+
+                # Fantasy models
+                target_x = torch.tensor([[0.5]])
+                # TODO: make this more efficient
+                sampler = SobolQMCNormalSampler(1000, seed=9876)
+                fantasy_models = model.fantasize(target_x, sampler=sampler)
```
Comment (Balandat) [inline test/models/test_model_list_gp_regression.py:340]: Actually doing an MC estimate here is nice in the sense that it tests this code e2e, but as you say it's not very efficient.

One alternative to comparing the statistics here would be to just mock things. E.g. one idea would be to just mock the `get_fantasy_model` call [here](https://github.com/pytorch/botorch/blob/main/botorch/models/model_list_gp_regression.py#L98) and then check that the passed in `targets` are actually transformed versions of the ones passed into `condition_on_observations`. 

Or you could mock even more and just check that every model in the list that has an outcome transform ha the outcome transform called on the appropriate index of the passed in samples `Y`.

File: test/models/test_model_list_gp_regression.py [hunk 2]
```diff
@@ -312,3 +312,69 @@ def test_fantasize(self):
...
+            FixedNoiseGP(X, Y, yvar, outcome_transform=Standardize(m=1))
+        )
+
+        model.posterior(torch.zeros((1, 1)))
+
+        fant = model.fantasize(X, sampler=IIDNormalSampler(n_fants, seed=0), noise=yvar)
+
+        fant_mean = fant.posterior(X).mean.mean(0).detach().numpy().flatten()
+        self.assertAlmostEqual(fant_mean[0], y_at_low_x, delta=5e-4)
+        self.assertAlmostEqual(fant_mean[1], y_at_high_x, delta=1e-3)
```
Comment (Balandat) [inline test/models/test_model_list_gp_regression.py:380]: I know you're fixing the seed here but I recall some issues with that not working reliably, did you run a stress test of this to make sure this won't be flaky here?

File: test/models/test_model_list_gp_regression.py [hunk 3]
```diff
@@ -312,3 +312,69 @@ def test_fantasize(self):
...
+        yvar = torch.full_like(Y, 1e-4)
+        model = ModelListGP(
+            FixedNoiseGP(X, Y, yvar, outcome_transform=Standardize(m=1))
+        )
+
+        model.posterior(torch.zeros((1, 1)))
+
+        fant = model.fantasize(X, sampler=IIDNormalSampler(n_fants, seed=0), noise=yvar)
+
+        fant_mean = fant.posterior(X).mean.mean(0).detach().numpy().flatten()
```
Comment (Balandat) [inline test/models/test_model_list_gp_regression.py:378]: No need to make into a numpy array, can just make it into a list of python scalars. 
```suggestion
        fant_mean = fant.posterior(X).mean.mean(0).flatten().tolist()
```

**Review by Balandat — APPROVED** [review]
Thanks, changes look good to me!

File: test/models/test_model_list_gp_regression.py (hunk 2 above)
Comment (esantorella) [reply test/models/test_model_list_gp_regression.py:380]: yeah, Actions says this is not working reliably

File: test/models/test_model_list_gp_regression.py (hunk 1 above)
Comment (esantorella) [reply test/models/test_model_list_gp_regression.py]: Yeah it would be nice to have some more e2e MC checks, but I don't love it as a unit test due to the inefficiency and high potential to become flaky from minor numerical issues.

**Review by Balandat — APPROVED** [review]
gtmtl, thanks!
